File: 1-week/set1/mario.c
#include <cs50.h>
#include <stdio.h>

void print_blocks(int size, int full_size);
bool valid_input(int size);
void init(void);

//gcc -g mario.c -lcs50 -o mario
int main(void)
{
    init();
}

/**
 * @brief Will initialize the project. Call methods to validate and print blocks
 * With this method, I'm able to call again the method instead to put a do while
 */
void init(void)
{
    int size = get_int("Height: ");
    bool is_valid = valid_input(size);
    int const c_size = size;
    // If valid, print the bocks
    if (is_valid == true)
    {
        int i = 0;
        while (size > 0)
        {
            print_blocks(i++, c_size);
            // Decrement size to loop
            size--;
            printf("\n");
        }
    }
    else
    {
        // Initialize again, until digit something correct
        init();
    }
}

/**
 * @brief Handle the print blocks
 * @param size Size to print block
 */
void print_blocks(int size, int full_size)
{
    // Fill by blank until full size. j start with 1
    int j = 1;
    while (j < (full_size - size))
    {
        // Fill with white space
        printf(" ");
        j++;
    }

    int i = 0;
    while (i <= size)
    {
        // Print Block
        printf("#");
        i++;
    }
}

/**
 * @brief Validate the input passed by user. Must be between 1 and 9
 * @param size - Size typed by user
 * @return boolean - Valid or invalid
 */
bool valid_input(int size)
{
    if (size >= 1 && size <= 8)
    {
        return (true);
    }
    else
    {
        return (false);
    }
}

File: 6-week/problem/sentimental-mario-less/mario.py
from cs50 import get_int


def main():
    init()


# Initialization method
def init():
    size = get_int("Height: ")
    is_valid = validate_input(size)
    c_size = size
    if is_valid == True:
        i = 0
        while size > 0:
            print_blocks(i, c_size)
            i += 1
            # Decrement size loop
            size -= 1
            # New break line
            print("")
    else:
        init()


# Print blocks
def print_blocks(size, full_size):
    j = 1
    while j < full_size - size:
        print(" ", end="")
        j += 1

    i = 0
    while i <= size:
        print("#", end="")
        i += 1


# Validate the input passed by user
def validate_input(size):
    if size >= 1 and size <= 8:
        return True
    else:
        return False


if __name__ == "__main__":
    main()
